src/services/bundleAnalysis/useBranchBundleSummary.tsx

```diff
@@ -127,28 +130,20 @@ const query = `query BranchBundleSummaryData(
   }
 }`
 
-export interface UseBranchBundleSummaryArgs {
+export interface BranchBundleSummaryQueryOptsArgs {
   provider: string
   owner: string
   repo: string
-  branch?: string
+  branch: string | null | undefined
```

Comment:
Why this instead of `branch?: string | null`?

Reply:
This is an attempt to be a bit more explicit with function args and their types. `?` represents an optional argument so TS doesn't complain if you never provide it. However, if you type it as above, TS will complain if you don't provide the value:

![Screenshot 2024-11-01 at 12 17 48](https://github.com/user-attachments/assets/479c0d44-f71d-4a98-8e9a-bd8900803e6b)

Reply:
Oh so you just want this to always be called with a branch param?

Reply:
Yes I would like all the arguments provided, this still makes allowances however if `branch` is undefined.

Reply:
Ok cool beans